### PR TITLE
Refactor creation of RasaModelData

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
Simplify creation of RasaModelData by reusing methods of `model_data_utils.py`.

closes https://github.com/RasaHQ/research/issues/100

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by tabergma** [comment]
/modeltest

```yml
include:
 - dataset: ["Carbon Bot"]
   config: ["all"]
```

File: rasa/nlu/classifiers/diet_classifier.py [hunk 1]
```diff
@@ -628,90 +627,91 @@ def _create_model_data(
...
-
-        features = defaultdict(lambda: defaultdict(list))
-        label_ids = []
-
-        for example in training_data:
-            if label_attribute is None or example.get(label_attribute):
-                text_features = self._extract_features(example, TEXT)
-                for feature_key, feature_value in text_features.items():
-                    features[TEXT][feature_key].append(feature_value)
+        attributes_to_consider = [TEXT, label_attribute]
```
Comment (Ghostvv) [inline rasa/nlu/classifiers/diet_classifier.py:632]: can't we. make diet only entity extractor?

File: rasa/nlu/classifiers/diet_classifier.py [hunk 2]
```diff
@@ -628,90 +627,91 @@ def _create_model_data(
...
+        )
 
-            # only add tag_ids during training
-            if training and self.component_config.get(ENTITY_RECOGNITION):
-                for tag_spec in self._entity_tag_specs:
-                    features[ENTITIES][tag_spec.tag_name].append(
-                        self._tag_ids_for_crf(example, tag_spec)
-                    )
+        # during training we store the tag ids for the different entity labels under
+        # the key ENTITIES. during prediction we do not have any tag ids.
```
Comment (Ghostvv) [inline rasa/nlu/classifiers/diet_classifier.py:655]: what would `attribute_data[ENTITIES]` contain during prediction?

File: rasa/nlu/classifiers/diet_classifier.py [hunk 3]
```diff
@@ -628,90 +627,91 @@ def _create_model_data(
...
+        if not training and ENTITIES in attribute_data:
+            del attribute_data[ENTITIES]
```
Comment (Ghostvv) [inline rasa/nlu/classifiers/diet_classifier.py:657]: does it make sense to move it to `model_data_utils.convert_to_data_format`?

File: rasa/utils/tensorflow/model_data_utils.py [hunk 4]
```diff
@@ -1,189 +1,281 @@
...
+    training_examples: List[Message],
+    attributes: List[Text],
+    entity_tag_specs: Optional[List["EntityTagSpec"]] = None,
+    featurizers: Optional[List[Text]] = None,
+    bilou_tagging: bool = False,
+) -> List[Dict[Text, List["Features"]]]:
+    """Converts training data into a list of attribute to features.
 
-MASK = "mask"
+    Possible attributes are, for example, INTENT, RESPONSE, TEXT, or ENTITIES.
```
Comment (Ghostvv) [inline rasa/utils/tensorflow/model_data_utils.py:35]: ACTION_TEXT and ACTION_NAME also?

File: rasa/nlu/classifiers/diet_classifier.py (hunk 2 above)
Comment (tabergma) [reply rasa/nlu/classifiers/diet_classifier.py:655]: It is basically an empty list, but we can just not add `ENTITIES` to the `attributes_to_consider` during prediction, that should be simpler, will update the code

File: rasa/nlu/classifiers/diet_classifier.py (hunk 3 above)
Comment (tabergma) [reply rasa/nlu/classifiers/diet_classifier.py:657]: not needed if you just add `ENTITIES` as attribute during training.

File: rasa/utils/tensorflow/model_data_utils.py (hunk 4 above)
Comment (tabergma) [reply rasa/utils/tensorflow/model_data_utils.py:35]: true, will add those

**Review by Ghostvv — APPROVED** [review]

**Pull request merged**